### PR TITLE
enable GitHub actions for building

### PR DESCRIPTION
**Pull request by vladak**
Travis backlog has started being unbearable recently (see https://www.traviscistatus.com/). This change enables basic Github actions for building. Relase related tasks will be done in later PR.

The macOS build fails due to #3296. For the Windows build to pass I had to exclude some of the tests. Will file issues for these.

Info about the build environments: https://github.com/actions/virtual-environments

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - name: Cache Maven packages
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Install Universal ctags
+      run: choco install universal-ctags
+    - name: Build
+      run: mvn -B -V verify
```
Comment (ahornace) [inline .github/workflows/build.yml:69]:
```suggestion
      run: ./mvnw -B -V verify
```
Maybe use `mvnw`?

Comment (vladak) [reply .github/workflows/build.yml:69]: I tried that, does not work as the default shell is Powershell. Maven is installed by default the build runs fine.

File: .github/workflows/javadoc.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      with:
+        java-version: 11
+    - name: Cache Maven packages
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Refresh Javadoc
+      run: ./dev/docker.sh
```
Comment (ahornace) [inline .github/workflows/javadoc.yml:22]:
```suggestion
      run: ./dev/javadoc.sh
```
Maybe this script instead of `docker.sh`?

**Comment by ahornace** [comment]
Why not remove javadoc and docker stages from travis as well?

**Comment by vladak** [comment]
> Why not remove javadoc and docker stages from travis as well?

Not yet - they contain release specific pieces. At this point I'd like to preserve the ability to create releases via Travis, even though it is slow.

**Review by ahornace — APPROVED** [review]

**Comment by vladak** [comment]
Thanks !

**Pull request merged**